### ISSUE TITLE
Remote Validation Only Works for DateTimes

### ISSUE THREAD

**Issue by c-wiens** [aspnet/Mvc#8544]
#### Is this a Bug or Feature request?: 
Bug

#### Steps to reproduce (preferably a link to a GitHub repo with a repro project):
Model:
```csharp
public partial class Model1
{
    ...
   [Remote(action: "DateTimeTest1", controller: "Home")]
    public DateTime myDate1 { get; set; }
    [Remote(action: "DateTimeTest2", controller: "Home")]
    public DateTime myDate2 { get; set; }
    [Remote(action: "StringTest", controller: "Home")]
    public string myString { get; set; }
    [Remote(action: "IntTest", controller: "Home")]
    public int myInt { get; set; }
    ...
}
```

Controller:
```csharp
   [Authorize]
    public class HomeController : BaseController
    {
         ...
        public IActionResult DateTimeTest1 (DateTime myDate1)
        {   // BREAKPOINT HERE IS HIT
            return Json(true);
        }
        public IActionResult DateTimeTest2 (DateTime myDate2)
        {   // BREAKPOINT HERE IS HIT
            return Json(true);
        }
        public IActionResult StringTest (string myString)
        {   // BREAKPOINT HERE IS NEVER HIT
            return Json(true);
        }
        public IActionResult IntTest (DateTime myInt)
        {   // BREAKPOINT HERE IS NEVER HIT
            return Json(true);
        }
        ...
    }
```

View:
```html
<script src="~/lib/jquery.min.js"></script>

<div class="form-group">
    <label asp-for="DateTimeTest1">DateTimeTest1: </label>
    <div class='input-group date' id='DatePicker'>
        <input asp-for="DateTimeTest1" asp-format="{0:MM/dd/yyyy}" class="form-control" type="text" />
        <span class="input-group-addon">
            <span class="glyphicon glyphicon-calendar"></span>
        </span>
    </div>
    <span asp-validation-for="DateTimeTest1" class="text-danger"></span>
</div>
...
... same thing for the rest of the Model attributes...

<script src="~/lib/jquery-validation/dist/jquery.validate.min.js"></script>
<script src="~/lib/jquery-validation/dist/additional-methods.min.js"></script>
<script src="~/lib/jquery-validation-unobtrusive/jquery.validate.unobtrusive.min.js"></script>
```

#### Description of the problem:
In one of my model classes I have several different attributes that use [Remote] for jquery validation. If I put a breakpoint in all my validation methods in a controller, only the ones having to do with _DateTime_ attributes are hit.
Model attributes that are _strings_ and _ints_ seem to be completely ignored.

#### Version of `Microsoft.AspNetCore.Mvc` or `Microsoft.AspNetCore.App` or `Microsoft.AspNetCore.All`:
Microsoft.AspNetCore.App 2.1.4

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @c-wiens.
@javiercn, can you please look into this? Thanks!

**Comment by javiercn** [comment]
@c-wiens I'm not able to reproduce this issue. There are small issues in your sample code, like the one below, that might cause not to work.

> public IActionResult IntTest (DateTime myInt) { // BREAKPOINT HERE IS NEVER HIT return Json(true); }
This is expecting an int.

Do you see the network request go off on the browser network tab?
Are there any errors on the console?

Could you provide a minimal project that reproduces the issue?
I haven't been able to reproduce it.

**Comment by mkArtakMSFT** [comment]
Hi. We're closing this issue as we have heard no response from you for some time now. If you have more details and are encountering this issue please add a new reply and re-open the issue.

**Comment by c-wiens** [comment]
Additional details:
When I deleted all the javascript except the unobtrusive validation scripts, everything started working correctly again.

The issue for my Int-type select boxes was that I was using a jquery plugin which hid the actual select box, and I had configured the validator to ignore hidden elements:
`$.validator.setDefaults({ ignore: ":hidden" });`

Once I set the validator to `$.validator.setDefaults({ ignore: ".validation-ignore" });` (and gave the elements that I want to ignore the validation-ignore class) everything worked for the IntTest method.

I'm still not sure what is keeping the StringTest from being hit.
